File: androidApp/src/main/java/com/commuto/interfacemobile/android/p2p/messages/TakerInformationMessage.kt
package com.commuto.interfacemobile.android.p2p.messages

import com.commuto.interfacemobile.android.key.keys.PublicKey
import java.util.*

/**
 * This represents a Taker Information Message, as described in the
 * [Interface Specification](https://github.com/jimmyneutront/commuto-whitepaper/blob/main/commuto-interface-specification.txt).
 *
 * @property swapID The ID of the swap for which taker information is being sent.
 * @property publicKey The taker's public key.
 * @property settlementMethodDetails The taker's settlement method details, or `null` if no such data exists.
 */
data class TakerInformationMessage(val swapID: UUID, val publicKey: PublicKey, val settlementMethodDetails: String?)

File: iosApp/iosApp/p2p/messages/TakerInformationMessage.swift
//
//  TakerInformationMessage.swift
//  iosApp
//
//  Created by jimmyt on 8/22/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation

/**
 This represents a Taker Information Message, as described in the [Interface Specification](https://github.com/jimmyneutront/commuto-whitepaper/blob/main/commuto-interface-specification.txt).
 */
struct TakerInformationMessage {
    /**
     The ID of the swap for which taker information is being sent.
     */
    let swapID: UUID
    /**
     The taker's public key.
     */
    let publicKey: PublicKey
    /**
     The taker's settlement method details.
     */
    let settlementMethodDetails: String
}
